netneurotools/metrics.py

```diff
@@ -99,3 +101,87 @@ def communicability_wei(adjacency):
     cmc[np.diag_indices_from(cmc)] = 0
 
     return cmc
+
+
+def rich_feeder_peripheral(x, sc, stat='median'):
+    """
+    Calculates median or mean "connectivity" in rich (hub to hub), feeder
+    (hub to non-hub), and peripheral (non-hub to non-hub) links.
+
+    This function takes a square symmetric correlation/connectivity matrix `x`
+    and computes the median or mean value within rich, feeder, and peripheral
+    links. Hubs are defined againt a threshold (from 0 to the maximum degree
+    `k`, defined on the structural connectivity matrix, `sc`).
+
+    Parameters
+    ----------
+    x : (N, N) numpy.ndarray
+        Symmetric correlation or connectivity matrix
+    sc : (N, N) numpy.ndarray
+        binary structural connectivity matrix
+    stat : str
+        statistic to use over rich/feeder/peripheral links
+        'mean' or 'median' (default)
+
+    Returns
+    -------
+    rfp : (3, k) numpy.ndarray
+        Array of median rich (0), feeder (1), and peripheral (2)
+        values, defined by `x`. `k` is the maximum degree defined on `sc`.
+    pvals : (3, k) numpy.ndarray
+        p-value for each link, computed using Welch's t-test.
+        Rich links are compared against non-rich links. Feeder links are
+        compared against peripheral links. Peripheral links are compared
+        against feeder links. T-test is two-sided.
+
+    Author
+    ------
+    This code was written by Justine Hansen who promises to fix and even
+    optimize the code should any issues arise, provided you let her know.
+    """
+
```

Comment:
It would be good to add a check up here to confirm that the provided `stat` is one of the acceptable values. You can do something like:

```python
    stats = ['mean', 'median']
    if stat not in stats:
        raise ValueError(f'Provided stat {stat} not valid. Must be one of {stats}')
```

(Or similar!)